src/edu/csus/ecs/pc2/clics/API202306/CLICSAccount.java

```diff
@@ -0,0 +1,320 @@
+        }
+        type = "" + cid.getClientType();
+        type = type.toLowerCase();
+        if(account.isTeam()) {
+            team_id = username;
+        }
+    }
+
+    public String toJSON() {
+
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            return mapper.writeValueAsString(this);
+        } catch (Exception e) {
+            return "Error creating JSON for account info " + e.getMessage();
+        }
+    }
+
+    /**
+     * Create account list from a accounts.json like file
+     *
+     * @param contest the contest (needed for groups)
+     * @param jsonfile json file to deserialize
+     * @param site the site to create the accounts for
+     * @return array of accounts to add, or null on error
+     */
+    public static Account [] fromJSON(IInternalContest contest, File jsonfile, int site) {
+        Account [] newaccounts = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            newaccounts = createAccountsFromCLICSAccounts(contest, mapper.readValue(jsonfile, CLICSAccount[].class), site, log);
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize account file " + jsonfile, e);
+        }
+        return(newaccounts);
+    }
+
+    /**
+     * Create account list from a accounts.yaml like file
+     *
+     * @param contest the contest (needed for groups)
+     * @param yamlfile yaml file to deserialize
+     * @param site the site to create the accounts for
+     * @return array of accounts to add, or null on error
+     */
+    public static Account [] fromYAML(IInternalContest contest, File yamlfile, int site) {
+        Account [] newaccounts = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+            newaccounts = createAccountsFromCLICSAccounts(contest, mapper.readValue(yamlfile, CLICSAccount[].class), site, log);
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize account file " + yamlfile, e);
+        }
+        return(newaccounts);
+     }
+
+    /**
+     * Create account list from a string containing accounts yaml
+     *
+     * @param contest the contest (needed for groups)
+     * @param yaml yaml to deserialize
+     * @param site the site to create the accounts for
+     * @return array of accounts to add, or null on error
+     */
+    public static Account [] fromJSON(IInternalContest contest, String json, int site) {
+        Account [] newaccounts = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            newaccounts = createAccountsFromCLICSAccounts(contest, mapper.readValue(json, CLICSAccount[].class), site, log);
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize team string", e);
```

Comment:
This log message refers to `team string`; I don't think that's correct.  (from Doug?)